grafana/resource_stack.go

```diff
@@ -0,0 +1,274 @@
+					value := v.(string)
+					// Only acceptable regions are eu and us
+					if value != "eu" && value != "us" {
+						errors = append(errors, fmt.Errorf("region '%s' is not supported. Only 'eu' and 'us' are currently supported", value))
+					}
+					return
+				},
+			},
+			"url": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				Description: "Custom URL for the Grafana instance. Must have a CNAME setup to point to `.grafana.net` before creating the stack",
+			},
+			"org_id": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Organization id to assign to this stack.",
+			},
+			"org_slug": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Organization slug to assign to this stack.",
+			},
+			"org_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Organization name to assign to this stack.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Status of the stack.",
+			},
+			"prometheus_user_id": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Promehteus user ID. Used for e.g. remote_write.",
+			},
+			"prometheus_url": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Prometheus url for this instance.",
+			},
+			"prometheus_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Prometheus name for this instance.",
+			},
+			"prometheus_remote_endpoint": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Use this URL to query hosted metrics data e.g. Prometheus data source in Grafana",
+			},
+			"prometheus_remote_write_endpoint": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Use this URL to send prometheus metrics to Grafana cloud",
+			},
+			"prometheus_status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Prometheus status for this instance.",
+			},
+			"alertmanager_user_id": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "User ID of the Alertmanager instance configured for this stack.",
+			},
+			"alertmanager_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the Alertmanager instance configured for this stack.",
+			},
+			"alertmanager_url": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Base URL of the Alertmanager instance configured for this stack.",
+			},
+			"alertmanager_status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Status of the Alertmanager instance configured for this stack.",
+			},
+		},
+	}
+}
+
+func CreateStack(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*client).gapi
```

Comment:
Since https://github.com/grafana/terraform-provider-grafana/pull/358, you can use
```suggestion
	client := meta.(*client).gcloudapi
```